### ISSUE TITLE
Button Group component (new)

### ISSUE THREAD

**Issue by reddolan** [vmware/clarity#247]
### Description
A Button Group is a collection of Buttons aligned side by side with no space between them. This can be used to create a collection of similar actions, to save space, or to respond to different size containers.

### Use cases under consideration
- mixing colored buttons for action differentiation
- adding dropdown menus for additional options
- a responsive experience to handle smaller containers
- icon buttons
- checkbox and radio group selections

#### Basic
- Normal Button design options are available to Button Groups as well
- Available for each type of button
- Group together similar types of actions that a user can do
- Can be used to maximize UI space

![basic](https://cloud.githubusercontent.com/assets/1858226/24614548/2143289e-1851-11e7-9258-0e1a1acd04b0.png)

#### Mixed Classes
- action types can be mixed in a Button Group
- we recommend that primary actions be the first in a group, followed by secondary actions
- we recommend that warning actions be the last in a group

![mixed-classes](https://cloud.githubusercontent.com/assets/1858226/24615320/8fac6c08-1853-11e7-805c-29ee2f8dcb76.png)

#### Icons
- icons used in Button Group will have minimum button width of 72px
- flat Buttons have no minimum width. There is only 24px space between icons
- flat Button Groups can Action Blue or Grey. Grey is used with Grid and Table actions

![icons](https://cloud.githubusercontent.com/assets/1858226/24615795/e748d342-1854-11e7-9975-943b2a1e0951.png)

#### Dropdowns
- used when a button has a set of actions related to it
- caret is to the right of the text or icon
- the dropdown will align to the left-most edge of the button or caret button in Split

![dropdowns](https://cloud.githubusercontent.com/assets/1858226/24615811/f02aa6d4-1854-11e7-826b-cfe4757939d6.png)

#### Overflow
- used for preserving space when many buttons are present
- can be mannually configured or set to automatically show based on space
- can be configured “x” number of buttons and “y” number of actions in the dropdown

![overflow](https://cloud.githubusercontent.com/assets/1858226/24615828/faa8bf06-1854-11e7-8a05-97d1d3f1e741.png)

#### Multiple Groups
- Button Groups next to one another
- use when you want similar actions to be grouped together and separated from others
- uses normal spacing between Buttons (12px)
- we recommend starting with primary actions followed by secondary actions

![multiple-groups](https://cloud.githubusercontent.com/assets/1858226/24615836/07905738-1855-11e7-8035-28e3125f2267.png)

#### Checkbox
- use when a small list of options can be selected from, similar to the Checkbox component
- blue indicates an option is selected, white (outline) indicates an unselected option
- user can click the button to select, and click again to deselect
- we recommend that an option label use 1-3 words
- can be configured with preselected options 

![checkbox](https://cloud.githubusercontent.com/assets/1858226/22164760/270b41e2-df1e-11e6-875a-e76e74a68bf4.png)

#### Radio
- use when selecting one option from a small list of options, similar to the Radio component
- blue indicates an option is selected, white (outline) indicates an unselected option
- selecting an unselected option, will deselect the previously selected option
- we recommend that an option label use 1-3 words
- can be configured with a preselected option

![radio](https://cloud.githubusercontent.com/assets/1858226/24615858/18d73764-1855-11e7-966c-b2de5d2561cf.png)

### References
- [https://v4-alpha.getbootstrap.com/components/button-group/](https://v4-alpha.getbootstrap.com/components/button-group/)
- [https://v4-alpha.getbootstrap.com/components/buttons/#checkbox-and-radio-buttons](https://v4-alpha.getbootstrap.com/components/buttons/#checkbox-and-radio-buttons)

**Documentation is subject to change**

**Comment by reddolan** [comment]
Original request - #240

**Comment by jaffoneh** [comment]
Closing for now as a duplicate of the original request.

**Comment by reddolan** [comment]
Updated with new documentation and suporting images

**Comment by vkpavan26** [comment]
I have query on the implementation/usage of this Button Group.

If this is applied on top of a datagrid,if we want same actions to be shown on right click on the datagrid as well will get the capability or we need to add one more control to handle that.?

Ex: of the similar usage is given below.
![image](https://cloud.githubusercontent.com/assets/25317393/22625259/385a6e82-ebb7-11e6-8403-dd11802b4e77.png)

**Comment by jaffoneh** [comment]
@vkpavan26 right-click implementation of actions on the datagrid wouldn't be there by default. You can add it by customizing the datagrid although the recommendation is to use the single-row click pattern recommended in #17 that is currently being implemented considering it works better in the context of touch screens as well (when thinking of touch screens we're not only talking about tablets or mobile phones, we're also talking about touch screen devices like Windows 10+ devices for example).

**Comment by vkpavan26** [comment]
Hi Jehad,

IMO grid should have that capability to expose the same actions on right click, so that whenever we are consuming the grid we need to add customization on our own in every implementation.

Even on single selection/multi selection we should have capability of both because there are usecases where you can allow user to remove multiple configurations in one go.

![image](https://cloud.githubusercontent.com/assets/25317393/22641341/5906abb2-ec7c-11e6-90d6-bb2540a0d59a.png)

**Comment by jaffoneh** [comment]
@vkpavan26 in terms of single selection/multi-selection, that does exist as part of #17 which is currently being implemented. We'll take the suggestion of right-click into consideration but as Clarity, we're trying to get away from using right-click as a main method of actions and moving forward with the design in #17. Based on that, we'll be able to gather more information once it is out and continue to iterate so feedback here is appreciated.

**Comment by halachliyskid** [comment]
@reddolan Does this design suggest that blue icons should be used in toolbars (e.g. datagrid toolbar)? I thought the intent was to have them grey (#565656) and then they'd turn blue on hover?

**Comment by reddolan** [comment]
@halachliyskid it's spec'd to have grey over datagrid. But you can have either grey or blue if you prefer. The documentation is in the Icons section above
> flat Button Groups can Action Blue or Grey. Grey is used with Grid and Table actions

**Comment by Jinnie** [comment]
Hi all,
Do you intend to allow stacking button with input fields, is it is possible with the [Bootstrap input-groups](https://v4-alpha.getbootstrap.com/components/input-group/)?

**Comment by reddolan** [comment]
@Jinnie that would be outside the scope of this component. However, if you need that type of component, please file a feature request issue for it.

**Comment by Jinnie** [comment]
@reddolan 
Not really. Just curious.
While we were using Bootstrap, we were using both button groups and input groups and the team seemed to like them.
Then we dropped Bootstrap and the groups were gone.
When I saw your work on the button groups, I thought that input groups may be coming next :) 
For me, personally, it is something cool, but the actual value is more moderate. I might use it if it exists, but I'd not insist for it or start implementing it myself.
